cmake-edit-ui/src/main/java/ru/urururu/cmakeedit/ui/CmakeTextPane.java

```diff
@@ -53,70 +55,28 @@
 
         if (text != null) {
             textPane.replaceSelection(text);
-            parseAll();
+            parseAll(true);
         }
 
         styledDocument.addDocumentListener(this);
     }
 
-    private void parseAll() {
-        styledDocument.setCharacterAttributes(0, styledDocument.getLength(), normal, true);
-        textPane.getHighlighter().removeAllHighlights();
-
-        FileNode fileNode;
-        try {
-            fileNode = Parser.parse(new DocumentParseContext(styledDocument), Parser.ErrorHandling.NodesBefore);
-        } catch (ParseException e) {
-            // shouldn't be reachable, we're returning partially parsed tree.
-            throw new IllegalStateException(e);
+    private void parseAll(boolean force) {
+        if (!(isDirty || force)) {
+            return;
         }
-
-        fileNode.visitAll(new NodeVisitorAdapter() {
-            @Override
-            public void accept(CommandInvocationNode node) {
-                colorize(node, normal);
-                for (Node child : node.getComments()) {
-                    child.visit(this);
-                }
-                for (Node child : node.getArguments()) {
-                    child.visit(this);
-                }
-            }
-
-            @Override
-            public void accept(ArgumentNode node) {
-                colorize(node, argument);
-                for (Node child : node.getChildren()) {
-                    child.visit(this);
-                }
-            }
-
-            @Override
-            public void accept(ExpressionNode node) {
-                colorize(node, expression);
-            }
-
-            @Override
-            public void accept(CommentNode node) {
-                colorize(node, comment);
-            }
-
-            @Override
-            public void accept(ParseErrorNode node) {
-                addHighlight(node.getStart(), node.getEnd(), errorsHighlighter);
-            }
-        });
-
-        try {
-            Checker.findUnused(new FileCheckContext(fileNode, new MetricRegistry(),(range, problem) -> addHighlight(range.getStart(), range.getEnd(), warningsHighlighter)));
-        } catch (LogicalException e) {
-            addHighlight(e.getFirstNode().getStart(), e.getLastNode().getEnd(), errorsHighlighter);
+        if (worker != null) {
+            worker.cancel(true);
         }
+
+        worker = new CheckerWorker();
+        worker.execute();
+        isDirty = false;
     }
 
-    private void addHighlight(SourceRef start, SourceRef end, Highlighter.HighlightPainter warningsHighlighter) {
+    private void addHighlight(SourceRef start, SourceRef end, Highlighter.HighlightPainter painter) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Move this method into "CheckerWorker". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS3398)